packages/core/addon/serializers/dashboard.js

```diff
@@ -4,7 +4,45 @@
  */
 
 import AssetSerializer from 'navi-core/serializers/asset';
-import { get } from '@ember/object';
+import { getDefaultDataSourceName } from 'navi-data/utils/adapter';
+import { getOwner } from '@ember/application';
+
+function v1ToV2Filter(filter, metadataService) {
+  let source = undefined;
+  let dimension = undefined; // the filter.dimension can be stored as dataSource.dimension
+  if (filter.dimension.includes('.')) {
+    [source, ...dimension] = filter.dimension.split('.');
+    dimension = dimension.join('.');
+  } else {
+    dimension = filter.dimension;
+  }
+  const matchesField = d => d.id === dimension;
+  const matchingDimensions = metadataService.all('dimension', source).filter(matchesField);
+  const matchingTimeDimensions = metadataService.all('timeDimension', source).filter(matchesField);
```

Comment:
Do we think there is a case for multiple matches for a source? Could we use `getById`?